src/openfermion/transforms/_code_transform_functions.py

```diff
@@ -0,0 +1,254 @@
+    state. A Checksum code with odd weight will encode all states with odd
+    occupation number. This code saves one qubit: n_qubits = n_modes - 1.
+        
+    Args:
+        modes (int): number of modes
+        odd (int or bool): 1 (True) or 0 (False), if odd,
+            we encode all states with odd Hamming weight
+
+    Returns (BinaryCode): The checksum BinaryCode
+    """
+    return BinaryCode(_encoder_checksum(modes), _decoder_checksum(modes, odd))
+
+
+def jordan_wigner_code(modes):
+    """ The Jordan-Wigner transform as binary code.
+        
+    Args:
+        modes (int): number of modes
+
+    Returns (BinaryCode): The Jordan-Wigner BinaryCode
+    """
+    return BinaryCode(numpy.identity(modes, dtype=int), linearize_decoder(
+        numpy.identity(modes, dtype=int)))
+
+
+def bravyi_kitaev_code(modes):
+    """ The Bravyi-Kitaev transform as binary code. The implementation
+    follows arXiv:1208.5986.
+        
+    Args:
+        modes (int): number of modes
+
+    Returns (BinaryCode): The Bravyi-Kitaev BinaryCode
+    """
+    return BinaryCode(_encoder_bk(modes),
+                      linearize_decoder(_decoder_bk(modes)))
+
+
+def parity_code(modes):
+    """ The parity transform (arXiv:1208.5986) as binary code. This code is
+    very similar to the Jordan-Wigner transform, but with long update strings
+    instead of parity strings. It does not save qubits: n_qubits = n_modes.
+        
+    Args:
+        modes (int): number of modes
+
+    Returns (BinaryCode): The parity transform BinaryCode
+    """
+    dec_mtx = numpy.reshape(([1] + [0] * (modes - 1)) +
+                            ([1, 1] + (modes - 1) * [0]) * (modes - 2) +
+                            [1, 1], (modes, modes))
+    enc_mtx = numpy.tril(numpy.ones((modes, modes), dtype=int))
+
+    return BinaryCode(enc_mtx, linearize_decoder(dec_mtx))
+
+
+def weight_one_binary_addressing_code(exponent):
+    """ Weight-1 binary addressing code (arXiv:1712.07067). This highly
+    non-linear code works for a number of modes that is an integer power
+    of two. It encodes all possible vectors with Hamming weight 1, which
+    corresponds to all states with total occupation one. The amount of
+    qubits saved here is maximal: for a given argument 'exponent', we find
+    n_modes = 2 ^ exponent, n_qubits = exponent. 
+
+    Note:
+        This code is highly non-linear and might produce a lot of terms.
+
+    Args:
+        exponent (int): exponent for the number of modes n_modes = 2 ^ exponent
+
+    Returns (BinaryCode): the weight one binary addressing BinaryCode
+    """
+    encoder = numpy.zeros((exponent, 2 ** exponent), dtype=int)
+    decoder = [0] * (2 ** exponent)
+    for counter in numpy.arange(2 ** exponent):
+        encoder[:, counter], decoder[counter] = \
+            _binary_address(exponent, counter)
+    return BinaryCode(encoder, decoder)
+
+
+def weight_one_segment_code():
```

Comment:
I'm a little confused why this function that only works for 2-qubits is necessary.